### PR TITLE
A more conservative platform header; just win

### PR DESCRIPTION
**Pull request by bkentel**
- This is an updated "centralized-ifdefs"
- See #10806
- Tested under MSVC curses + SDL
- Tested under mingw-w64 curses + SDL

### CODE REVIEW

File: CataclysmWin.cbp [hunk 1]
```diff
@@ -381,6 +381,7 @@
 		<Unit filename="src/path_info.h" />
 		<Unit filename="src/pickup.cpp" />
 		<Unit filename="src/pickup.h" />
+        <Unit filename="src/platform_win.h" />
```
Comment (KA101) [inline CataclysmWin.cbp]: Tabba Tabba Hey!

File: src/platform_win.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#if defined(_MSC_VER)
+#   define CATA_COMPILER_MSVC _MSC_VER
```
Comment (kevingranade) [inline src/platform_win.h]: Quoting myself from the previous PR:
> I don't find the aliases for the platform defines to be an improvement, they just trade a standard define for a nonstandard one most of the time, and a pair of standard defines for a nonstandard one in a few places.

Just bundle up the windows-specific includes, don't alias the platform defines.
As a result of that, you're including this header in 17 files where it is adding negligible benefit, polluting those files with the (gigantic) windows headers.

**Comment by bkentel** [comment]
This does not include windows.h. It includes a small (maybe 200 line standalone) header that define symbolic names for the carious versions of windows.

**Comment by kevingranade** [comment]
I stand corrected, but that doesn't change the outcome.
Also, why not? Everywhere you need anything but the platform defines, it's also including windows.h already.

**Comment by bkentel** [comment]
I hadn't originally planned it to be used for just locations where windows.h was used, but I guess I'll make that change and remove the define aliases.

**Comment by bkentel** [comment]
Modified per @kevingranade 's last comment.

**Pull request merged**